website/blog/2021-02-19-tungsten.md

```diff
@@ -0,0 +1,509 @@
+
+To use sbt as a BSP server users still need to use `metals.bsp-switch` command,
+which will take care of setting up everything and adding the appropriate
+plugins.
+
+More information about sbt and BSP can be found in the previously published
+[blogpost](/metals/blog/2020/11/06/sbt-BSP-support).
+
+## [Scala 3] Support for Scala 3.0.0-RC1
+
+Scala `3.0.0-RC1` has just been released and it's the first release that starts
+us on the way towards the full Scala 3 release. Syntax is now stable and the
+compiler team will now focus on fixing existing issues and polishing the
+internals.
+
+This is a perfect opportunity to try out Scala 3 alongside Metals and see all
+the new and interesting features that the language has to offer as well as
+report any issues.
+
+## [Scala 3] Filtering and sorting for completions.
+
+In previous versions of Metals, the completions offered by the language server
+would just be a list of what is available in the current scope. This however did
+not provide the best possible experience to the users. Thanks to work by
+[@sellophane](https://github.com/sellophane) those initial results are now
+filtered and sorted so that the most useful methods and values will be available
+first.
+
+## [Scala 3] Basic shorter signature support for hover.
+
+Together with better completions we also included a better way to display types
+on hover. Previously, we would always display the full class name with the
+package prefix, which would mean that some methods could look quite complicated
+for example:
+
+`def method(first: alpha.beta.gamma.DType, second: alpha.beta.gamma.DType)`
+
+Thanks to [@tanishiking](https://github.com/tanishiking) we no longer display
+the long type for anything that is already imported into scope. So the above
+method would look like:
+
+`def method(first: DType, second: DType)`
+
+This needs to be further improved by shortening some of the symbols that are not
+imported into scope and we still need to apply the same optimization in other
+places where the type is displayed.
+
+## [Scala 3] Parser-based features for Scala 3
+
+Under the hood Metals uses a separate parser implementation that can be found in
+[scalameta](https://github.com/scalameta/scalameta) and it supports numerous
+different Scala versions. Thanks to the recent work on supporting Scala 3, we
+now have a dialect that should parse the Scala 3 syntax correctly on par with
+3.0.0-RC1 version.
+
+This parser is used by Metals to provide a number of different features, that
+should now work correctly for Scala 3:
+
+- document outline which shows the map for the current source file and can also
+  be searched, which is useful to locate a particular method or value in the
+  current document.
+- folding ranges which enables users to hide parts of their code.
+- automatic management of pipes in multiline string.
+- code action to convert simple string into multiline string.
+- go to definition in Scala 3 dependency sources.
```

Comment:
I added this point and the sentence below to mention the recently merged improvements around navigating dependency sources and generating semanticdb for Scala 3.